src/main/java/appeng/client/gui/implementations/GuiInterfaceTerminal.java

```diff
@@ -584,16 +585,14 @@ private boolean itemStackMatchesSearchTerm(final ItemStack itemStack, final Stri
                         .getItemDisplayName(AEApi.instance().storage().getStorageChannel(IItemStorageChannel.class).createStack(parsedItemStack))
                         .toLowerCase();
 
-                for (String term : searchTerm.split(" ")) {
+                for (String term : splitTerm) {
                     if (term.length() > 1 && (term.startsWith("-") || term.startsWith("!"))) {
                         term = term.substring(1);
                         if (displayName.contains(term)) {
                             return false;
                         }
                     } else if (displayName.contains(term)) {
                         foundMatchingItemStack = true;
```

Comment:
Should this not just return early?

Reply:
We don't want to return early because a later statement could hit the negate clause, i.e. `wood !oak`